statistics.py

```diff
@@ -62,3 +62,71 @@ def print_average_messages_length_by_sender(chat: Chat) -> None:
             total_messages_length_by_sender[sender] / messages_count_by_sender[sender], 2
         )
         print(f"The average len of {sender.name}'s message is {average_message_length} symbols")
+
+
+def _get_replies_by_users(chat: Chat) -> Dict[User, Dict[User, int]]:
+    replies_by_users: Dict[User, Dict[User, int]] = dict()
+    for message in chat.messages:
+        if not message.reply_to_message_id:
+            continue
+        sender = message.sender
+        replied_message = chat.get_message(message.reply_to_message_id)
+        if sender not in replies_by_users:
+            replies_by_users[sender] = dict()
+        if replied_message.sender not in replies_by_users[sender]:
+            replies_by_users[sender][replied_message.sender] = 0
+        replies_by_users[sender][replied_message.sender] += 1
+
+    return replies_by_users
+
+
+@statistic(name="most often replies")
+def print_most_often_replies(chat: Chat) -> None:
+    replies_by_users = _get_replies_by_users(chat)
+    for sender in replies_by_users:
+        first_replied_name = second_replied_name = third_replied_name = ""
+        first_number_of_replies = second_number_of_replies = third_number_of_replies = 0
+        for replied_user in replies_by_users[sender]:
+            if replies_by_users[sender][replied_user] > first_number_of_replies:
+                first_number_of_replies = replies_by_users[sender][replied_user]
+                first_replied_name = replied_user.name
+            elif replies_by_users[sender][replied_user] > second_number_of_replies:
+                second_number_of_replies = replies_by_users[sender][replied_user]
+                second_replied_name = replied_user.name
+            elif replies_by_users[sender][replied_user] > third_number_of_replies:
+                third_number_of_replies = replies_by_users[sender][replied_user]
+                third_replied_name = replied_user.name
+        if first_number_of_replies == 0:
+            print(f"{sender.name} replies nobody", end="")
+        else:
+            print(f"{sender.name} most often replies {first_replied_name} ({first_number_of_replies} times)", end="")
+        if second_number_of_replies > 0:
+            print(f", {second_replied_name} ({second_number_of_replies} times)", end="")
+        if third_number_of_replies > 0:
+            print(f", {third_replied_name} ({third_number_of_replies} times)", end="")
+        print()
+
```

Comment:
Ужасный кусок кода) Почему нельзя было сделать как-нибудь так? В следующей функции видимо такая же проблема. Сделай что-нибудь аналогичное

```suggestion
        top_replies = list(replies_by_users[sender].items())
        top_replies.sort(key=lambda x: -x[1])
        top_replies = top_replies[:3]

        if len(top_replies) == 0:
            print(f"{sender.name} replies nobody", end="")
        else:
            print(
                f"{sender.name} most often replies",
                ", ".join([f"{entry[0].name} ({entry[1]} times)" for entry in top_replies])
            )
```